test/test.go

```diff
@@ -20,48 +20,19 @@ import (
 )
 
 var (
-	exeSuffix string // ".exe" on Windows
+	ExeSuffix string // ".exe" on Windows
 	mu        sync.Mutex
 )
 
 func init() {
 	switch runtime.GOOS {
 	case "windows":
-		exeSuffix = ".exe"
+		ExeSuffix = ".exe"
 	}
 }
 
-// The TestMain function creates a dep command for testing purposes and
-// deletes it after the tests have been run.
-// Most of this is taken from https://github.com/golang/go/blob/master/src/cmd/go/go_test.go and reused here.
-func TestMain(m *testing.M) {
-	args := []string{"build", "-o", "testdep" + exeSuffix}
-	out, err := exec.Command("go", args...).CombinedOutput()
-	if err != nil {
-		fmt.Fprintf(os.Stderr, "building testdep failed: %v\n%s", err, out)
-		os.Exit(2)
-	}
-
-	// Don't let these environment variables confuse the test.
-	os.Unsetenv("GOPATH")
-	os.Unsetenv("GIT_ALLOW_PROTOCOL")
-	if home, ccacheDir := os.Getenv("HOME"), os.Getenv("CCACHE_DIR"); home != "" && ccacheDir == "" {
-		// On some systems the default C compiler is ccache.
-		// Setting HOME to a non-existent directory will break
-		// those systems.  Set CCACHE_DIR to cope.  Issue 17668.
-		os.Setenv("CCACHE_DIR", filepath.Join(home, ".ccache"))
-	}
-	os.Setenv("HOME", "/test-dep-home-does-not-exist")
-
-	r := m.Run()
-
-	os.Remove("testdep" + exeSuffix)
-
-	os.Exit(r)
-}
-
 // Manage a single run of the testgo binary.
-type testgoData struct {
+type TestgoData struct {
```

Comment:
This is now stuttering, should be `GoData`. Looking through what this does though it should probably be renamed anyway to something like `test.Helper` or something. Not sure if either should be done in this PR, or a follow up PR though.